### PR TITLE
Improve the help text of the LOADFIX command (#3221)

### PR DESCRIPTION
**Pull request by japsmits**
# Description

Improve the help text of the LOADFIX command

## Related issues

#3221 

# Manual testing

Manually ran loadfix/? to verify its output.

# Checklist

_Please tick the items as you have addressed them. Don't remove items; leave the ones that are not applicable unchecked._

I have:

- [x] followed the project's [contributing guidelines](https://github.com/dosbox-staging/dosbox-staging/blob/master/CONTRIBUTING.md) and [code of conduct](https://github.com/dosbox-staging/dosbox-staging/blob/master/CODE_OF_CONDUCT.md).
- [x] performed a self-review of my code.
- [ ] commented on the particularly hard-to-understand areas of my code.
- [x] split my work into well-defined, bisectable commits, and I [named my commits well](https://github.com/dosbox-staging/dosbox-staging/blob/main/CONTRIBUTING.md#commit-messages).
- [ ] applied the appropriate labels (bug, enhancement, refactoring, documentation, etc.)
- [ ] [checked](https://github.com/dosbox-staging/dosbox-staging/blob/main/scripts/compile_commits.sh) that all my commits can be built.
- [ ] confirmed that my code does not cause performance regressions (e.g., by running the Quake benchmark).
- [ ] added unit tests where applicable to prove the correctness of my code and to avoid future regressions.
- [ ] made corresponding changes to the documentation or the website according to the [documentation guidelines](https://github.com/dosbox-staging/dosbox-staging/blob/main/DOCUMENTATION.md).
- [ ] [locally verified](https://github.com/dosbox-staging/dosbox-staging/blob/main/DOCUMENTATION.md#previewing-documentation-changes-locally) my website or documentation changes.

### CODE REVIEW

**Review by weirddan455 — COMMENTED** [review]
One suggestion to clean up wording a bit.  The rest of this looks fine.

@johnnovak what are your thoughts on format commits?  I tend to run the `scripts/format-commit.sh` which only re-formats the part of the file that I actually touch.

The argument against re-formatting entire files is that it pollutes `git blame` and such for regression tracking.

**Comment by johnnovak** [comment]
> One suggestion to clean up wording a bit. The rest of this looks fine.
> 
> @johnnovak what are your thoughts on format commits? I tend to run the `scripts/format-commit.sh` which only re-formats the part of the file that I actually touch.
> 
> The argument against re-formatting entire files is that it pollutes `git blame` and such for regression tracking.

I would only reformat a file if I'm doing extensive changes to the whole thing, or at least most of it (so more than 60-70%). Otherwise I'm restricting the reformatting only to the parts I've touched (function level, sometimes even just block level in bigger functions). I'm using vim for that; just select the lines and run it through clang-format via a vim command.

I guess this would be my preference in general.

File: src/dos/program_loadfix.cpp [hunk 1]
```diff
@@ -90,22 +90,27 @@ void LOADFIX::AddMessages() {
...
+	        "  [color=light-cyan]GAME[reset]\t\tgame or program to load, optionally with parameters\n"
+	        "  [color=white]-SIZE[reset]\t\tSIZE indicates the number of kilobytes to be allocated\n"
+	        "  /d (or /f)\tFrees the previously allocated memory."
 	        "\n"
 	        "Notes:\n"
 	        "  - The most common use cases of this command are to fix DOS games or programs\n"
 	        "    which show either the \"[color=white]Packed File Corrupt[reset]\" or [color=white]\"Not enough memory\"[reset] (e.g.,\n"
-	        "    from some 1980's games such as California Games II) error message when run.\n"
+	        "    from some 1980's games such as California Games II and Wing Commander 2)\n"
+	        "    error message when run.\n"
```
[the remark this answers is not in the document]
Comment (japsmits) [reply src/dos/program_loadfix.cpp]: That sounds better. I just added Wing Commander 2 to the existing phrase, but it's good to make things readable. :smile.

**Comment by japsmits** [comment]
> > One suggestion to clean up wording a bit. The rest of this looks fine.
> > @johnnovak what are your thoughts on format commits? I tend to run the `scripts/format-commit.sh` which only re-formats the part of the file that I actually touch.
> > The argument against re-formatting entire files is that it pollutes `git blame` and such for regression tracking.
> 
> I would only reformat a file if I'm doing extensive changes to the whole thing, or at least most of it (so more than 60-70%). Otherwise I'm restricting the reformatting only to the parts I've touched (function level, sometimes even just block level in bigger functions). I'm using vim for that; just select the lines and run it through clang-format via a vim command.
> 
> I guess this would be my preference in general.

👍 I've removed the formatting commit, since it bigger than the actual work.

**Review by weirddan455 — APPROVED** [review]
Thanks.  Looks good now.  CI is still running so I'll check back and merge tomorrow probably.

File: src/dos/program_loadfix.cpp [hunk 2]
```diff
@@ -90,22 +90,26 @@ void LOADFIX::AddMessages() {
...
 	        "Usage:\n"
-	        "  [color=light-green]loadfix[reset] [color=light-cyan]GAME[reset] [color=white][PARAMETERS][reset]\n"
+	        "  [color=light-green]loadfix[reset] [color=white][-SIZE][reset] [color=light-cyan]GAME[reset] [color=white][PARAMETERS][reset]\n"
 	        "  [color=light-green]loadfix[reset] [/d] (or [/f])[reset]\n"
 	        "\n"
 	        "Parameters:\n"
-	        "  [color=light-cyan]GAME[reset]  game or program to load, optionally with parameters\n"
+	        "  [color=light-cyan]GAME[reset]\t\tgame or program to load, optionally with parameters\n"
+	        "  [color=white]-SIZE[reset]\t\tSIZE indicates the number of kilobytes to be allocated\n"
+	        "  /d (or /f)\tFrees the previously allocated memory."
```
Comment (johnnovak) [inline src/dos/program_loadfix.cpp:99]: We don't do tabs (or at least we shouldn't 😅). Can you please change these to spaces.

File: src/dos/program_loadfix.cpp [hunk 3]
```diff
@@ -90,22 +90,26 @@ void LOADFIX::AddMessages() {
...
 	        "\n"
 	        "Notes:\n"
-	        "  - The most common use cases of this command are to fix DOS games or programs\n"
-	        "    which show either the \"[color=white]Packed File Corrupt[reset]\" or [color=white]\"Not enough memory\"[reset] (e.g.,\n"
-	        "    from some 1980's games such as California Games II) error message when run.\n"
+	        "  - The most common use case of this command is to fix DOS games such as\n"
```
Comment (johnnovak) [inline src/dos/program_loadfix.cpp:102]: minor: Of course DOS games, as this is DOSBox. So I'd just say "games" 😄

**Comment by johnnovak** [comment]
> Thanks. Looks good now. CI is still running so I'll check back and merge tomorrow probably.

@weirddan455 Could you set the appropriate labels and the author before merging? Contributors can't do that, so it's our job for these PRs.

Labelling the PRs is important for generating the release notes and searchability, among other things.

File: src/dos/program_loadfix.cpp (hunk 3 above)
Comment (japsmits) [reply src/dos/program_loadfix.cpp:102]: It also runs WordPerfect 5.1 (or WordStar, for that matter), doesn't it? :wink: I would not be surprised if DOSBox is also used to run legacy DOS-programs under 64-bit Windows.

**Review by johnnovak — APPROVED** [review]
Looking good @japsmits. Congrats on your first approved PR 🎉 😄

**Comment by weirddan455** [comment]
@japsmits has gotten a PR merged before:  https://github.com/dosbox-staging/dosbox-staging/pull/3369

This will make number 2 :smile:

**Pull request merged**